transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala

```diff
@@ -279,34 +295,44 @@ object HcaPipelineBuilder extends PipelineBuilder[Args] {
           case Some(schema) => {
             // if the schema is not able to load, throw an exception, otherwise try to use it to validate
             Schema.loadFromString(schema) match {
-              case Failure(_) =>
-                throw new Exception(
-                  s"Schema not loaded properly for schema at $url, file $filename"
-                )
+              case Failure(_) => {
+                val errorMessage = s"Schema not loaded properly for schema at $url, file $filename"
+                logSchemaValidationError(filename, errorMessage)
+                throw new Exception(errorMessage)
```

Comment:
The fact that this throws at all is probably not very scala-like. Since the code is operating on a collection, having it return a collection of Either[(String, Msg), ValidationError] would be more what I'd expect, so each returned element would either be the validated tuple or the error for that tuple. But that kind of change depends a lot more on the context in which this is used.

Reply:
You're right, it should return something like that, or some type of `Option` or other thing listed here: https://docs.scala-lang.org/overviews/scala-book/functional-error-handling.html

Reply:
Part of the issue is that each successive step is already returning some kind of `Try` or `Either`; for example, `Schema.loadFromString` returns the type `Try[Schema]`, which needs to be matched on and dug into before moving on to the next step. I'm still upping my functional programming game, but my understanding is that Scala handles Exceptions via pattern matching. I wonder if there is a way to cleanly move all this matching out, and to make sure that the error logging only takes place at a `catch` equivalent.